src/core/memory.cpp

```diff
@@ -53,6 +53,9 @@ PAddr MemoryManager::Allocate(PAddr search_start, PAddr search_end, size_t size,
     PAddr free_addr = dmem_area->second.base;
     free_addr = alignment > 0 ? Common::AlignUp(free_addr, alignment) : free_addr;
 
+    // Align size
+    size = alignment > 0 ? Common::AlignUp(size, alignment) : size;
```

Comment:
I'm not sure if the size must be aligned too.

Reply:
Unless we can discover the relevant code, it is presumed to be okay. If a request result in too much allocated, it can be modified as necessary.

Reply:
I've taken a look at the kernel and the size is not aligned, rather it expects it to be aligned already.